ctowasm/src/processor/processExpression.ts

```diff
@@ -270,6 +270,24 @@ export default function processExpression(
           ...returnObjectMemoryLoads.slice(1),
         ],
       };
+    } else if (expr.type === "TypeCastingExpression") {
+      // type casting is just a way to tell the compiler to treat an expression as another type
+      // no actual conversion is done here
+      const processedExpr = processExpression(expr.expr, symbolTable);
+      const targetDataType = expr.targetDataType;
+
+      return {
+        originalDataType: targetDataType,
```

Comment:
the `originalDataType` type here does not seem to be used anywhere. How does the type conversion (e.g. `int`-> `float`) actually work in that case?